jaxrs-api/src/main/java/javax/ws/rs/core/MediaType.java

```diff
@@ -367,7 +371,14 @@ public boolean equals(final Object obj) {
     @SuppressWarnings("UnnecessaryJavaDocLink")
     @Override
     public int hashCode() {
-        return (this.type.toLowerCase() + this.subtype.toLowerCase()).hashCode() + this.parameters.hashCode();
+        int h = this.hash;
+        if (h == 0) {
+            h = this.parameters.hashCode();
```

Comment:
The PR's title is wrong now, as now it is not only about memoization, it also is about using a different algorithm. You should separate the PR into two commits, one for the memoization, and one for the changed algorithm.

Reply:
Squashing will not help the IP bot. Looking at all the time it fails, I really that bot is simply buggy. The only solution you have is: When it hangs, push again. When it fails, replace full name by Github ID.